### PR TITLE
FIX-#7234: Deprecate HDK engine

### LINKED ISSUE #7234: HDK deprecation
We stopped supporting HDK and [its repo](https://github.com/intel-ai/hdk) is archived. There are no plans to resume work on it in future. That's why, we are deprecating HDK engine and will remove it completely in a future release of Modin.

### PR DESCRIPTION
**Pull request by YarShev**
## What do these changes do?

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #7234  
- [x] tests passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

File: modin/config/envvars.py [hunk 1]
```diff
@@ -253,6 +253,26 @@ def _get_default(cls) -> str:
             "Please refer to installation documentation page to install an engine"
         )
 
+    @classmethod
+    def get(cls) -> str:
```
Comment (anmyachev) [inline modin/config/envvars.py:257]: What about `StorageFormat.get() == 'Hdk'`?

File: modin/error_message.py [hunk 2]
```diff
@@ -36,7 +36,9 @@ def not_implemented(cls, message: str = "") -> NoReturn:
         )
 
     @classmethod
-    def single_warning(cls, message: str) -> None:
+    def single_warning(
+        cls, message: str, category: Union[type[Warning], None] = None
```
Comment (anmyachev) [inline modin/error_message.py:40]: ?
```suggestion
        cls, message: str, category: Optional[Warning] = None
```

File: modin/error_message.py [hunk 3]
```diff
@@ -36,7 +36,9 @@ def not_implemented(cls, message: str = "") -> NoReturn:
...
+    ) -> None:
         message_hash = hash(message)
```
Comment (anmyachev) [inline modin/error_message.py:42]: I think we need to explicitly add a comment that there should not be identical messages with different categories, so only the message is used as the hash key.

Comment (YarShev) [reply modin/error_message.py:44]: added

**Review by anmyachev — APPROVED** [review]

**Pull request merged**